controllers/manilaapi_controller.go

```diff
@@ -223,49 +221,32 @@ func (r *ManilaAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			r.Log.Error(err, "Unable to retrieve API CRs %v")
 			return nil
 		}
-		for _, cr := range apis.Items {
-			for _, v := range cr.Spec.CustomServiceConfigSecrets {
-				if v == secretName {
+		// Watch for changes to secrets where the owner label AND the
+		// CR.Spec.ManagingCrName label matches
+		label := o.GetLabels()
+		if l, ok := label[labels.GetOwnerNameLabelSelector(labels.GetGroupLabel(manila.ServiceName))]; ok {
+			for _, cr := range apis.Items {
+				// return reconcile event for the CR where the owner label AND the parentCinderName matches
+				if l == manila.GetOwningManilaName(&cr) {
+					// return namespace and Name of CR
 					name := client.ObjectKey{
-						Namespace: namespace,
+						Namespace: o.GetNamespace(),
```

Comment:
might be a newbie go question, but wouldn't `Namespace: namespece` be okay too?

Reply:
This makes sense, yes, I didn't realize while moving / refactoring code that we already assigned `namespace = o.GetNamespace()`: good catch! Changing in my next iteration.

Reply:
Done